src/main/java/com/aws/greengrass/clientdevices/auth/configuration/MetricsConfiguration.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.configuration;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.util.Coerce;
+import lombok.Getter;
+
+import java.util.Optional;
+
+/**
+ * Represents the metrics part of the component configuration. Acts as an adapter from the GG topics to the domain.
+ * <p>
+ * |---- configuration
+ * |    |---- metrics:
+ * |          |---- emitMetrics
+ * |          |---- emittingFrequency
```

Comment:
The convention that we (Greengrass) have adopted for boolean values is that the "default" value should be false (think, the "zero-value" for a boolean variable). In this case, that would make it `disableMetrics`.

`period` is probably a more appropriate term than `frequency` in this context. `aggregationPeriod` perhaps?